rest/permissions.py

```diff
@@ -114,3 +118,88 @@ def has_permission(self, request, view):
                 RepoPermission.manage_taxonomy[0]
                 in get_perms(request.user, repo)
             )
+
+
+class AddEditMetadataPermission(BasePermission):
+    """Checks add_edit_metadata permission"""
+
+    def has_permission(self, request, view):
+        if request.method in SAFE_METHODS:
+            return True
+
+        try:
+            repo = get_repo(view.kwargs['repo_slug'], request.user.id)
+        except NotFound:
+            raise Http404()
+        except PermissionDenied:
+            return False
+
+        return (
+            RepoPermission.add_edit_metadata[0]
+            in get_perms(request.user, repo)
+        )
+
+
+class ViewLearningResourcePermission(BasePermission):
+    """
+    Checks that user has view_repo permission for the repository
+    and that that repo owns the learning resource being requested
+    """
+
+    def has_permission(self, request, view):
+        repo_slug = view.kwargs['repo_slug']
+        try:
+            get_repo(repo_slug, request.user.id)
+        except NotFound:
+            raise Http404()
+        except PermissionDenied:
+            return False
+
+        lr_id = int(view.kwargs['lr_id'])
+
+        try:
+            learning_resource = LearningResource.objects.get(id=lr_id)
+        except LearningResource.DoesNotExist:
+            raise Http404()
+
+        if learning_resource.course.repository.slug != repo_slug:
+            raise Http404()
+        return True
+
+
+class ViewStaticAssetPermission(BasePermission):
+    """
+    Checks that user has view_repo permission for the repository
+    and that that repo owns the learning resource being requested
+    """
+
+    def has_permission(self, request, view):
+        repo_slug = view.kwargs['repo_slug']
+        try:
+            get_repo(repo_slug, request.user.id)
+        except NotFound:
+            raise Http404()
+        except PermissionDenied:
+            return False
+
+        lr_id = int(view.kwargs['lr_id'])
+        sa_id = int(view.kwargs['sa_id'])
+
+        try:
+            learning_resource = LearningResource.objects.get(id=lr_id)
+        except LearningResource.DoesNotExist:
+            raise Http404()
+
+        if learning_resource.course.repository.slug != repo_slug:
+            raise Http404()
+
+        try:
```

Comment:
Since this duplicates `ViewLearningResourcePermission` exactly up to this point, is it possible to subclass it and just call `super()` within `has_permission`?